libwild/src/layout.rs

```diff
@@ -3608,9 +3608,11 @@ fn is_symbol_undefined(
         return false;
     }
 
-    sym_file_id == sym_def_file_id
+    let is_undefined = sym_file_id == sym_def_file_id
         && symbol.is_undefined(LittleEndian)
-        && symbol_value_flags.is_absolute()
+        && symbol_value_flags.is_absolute();
+
+    is_undefined && args.no_undefined
```

Comment:
The symbol is still undefined even if `no_undefined`is false. So I feel like the change to this line is changing the meaning of this function. Perhaps the function should be renamed? `should_emit_undefined_error`, `is_undefined_error`, `is_undefined_symbol_error` or something along those lines.